lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/UpdateTaskHost.cs

```diff
@@ -148,27 +155,80 @@ public byte[] Call(byte[] memento)
                     "Received OperationCanceledException in UpdateTaskHost with message: {0}.",
                     e.Message);
             }
-            catch (IOException e)
+            catch (Exception e)
             {
-                Logger.Log(Level.Error, "Received IOException in UpdateTaskHost with message: {0}.", e.Message);
-                if (!_cancellationSource.IsCancellationRequested)
+                if (e is IOException || e is TcpClientConnectionException || e is RemotingException ||
```

Comment:
Let's keep it as is for now, I think we'll have to clean up our exceptions separately at some point in (near) future